OutSystems/outsystems-datagrid-reactive
# ROU-3757: Fixing issue with undo/redo

### What was happening
* Whenever we searched for something, we were losing reference to the original rows, so when we undo/redo, the changes were reflected on the wrong row.

### What was done
* Adjusted undo/redo handlers to cancel if search is filled with text. 

### Test Steps
        1. Click on button "Add row"
        2. Edit the cell "Email-1" by typing "first"
        3.  Click on button "Mark Changes as saved"
        4. Click on button "Add row"
        5.  Edit the cell "Email-1" by typing "second"
        6. Fill search input with "first"
        7.  Select cell "Email-1"
        8. Press CtrlZ
        Expected: Cells "Email-1" is valid


### Screenshots
### Before
![validation](https://user-images.githubusercontent.com/3113318/192273935-8b226c77-75da-4081-b422-becabf8d393c.gif)

### After

![validation2](https://user-images.githubusercontent.com/3113318/192273967-79bbf902-9d3b-4477-8589-2d5765524237.gif)
